guardian/scala-steward-public-repos
# Supply default config to reduce daily PR noise

Some third-party libraries widely used at the Guardian (eg `software.amazon.awssdk`) automatically publish a new release _every single day_, often without obvious security benefit. Currently we don't _automatically_ merge the resulting daily Scala Steward PR, and so these low-value changes consume a lot of dev attention.

In general, we can accept a lower update cadence for these dependencies. Some individual Guardian repos have already introduced `.scala-steward.conf` config files to configure that cadence (see eg https://github.com/guardian/ophan/pull/4598), but to reduce duplication of effort, it would be good to have this lower cadence applied by default everywhere.

This update should configure our GitHub Action so that it will apply the default lower-cadence config to _all_ repos it runs on, rather than making devs individually configure it on every repo they work on.

The file location `.github/.scala-steward.conf` is the default value for [`repo-config` configuration input](https://github.com/scala-steward-org/scala-steward-action#configuration) on the Scala Steward Github Action. The support for default repo config in the Scala Steward Github Action was first introduced with https://github.com/scala-steward-org/scala-steward-action/pull/292, the name of the configuration input was later switched from `default-repo-conf` to `repo-config` with https://github.com/scala-steward-org/scala-steward-action/pull/324 .

### How to test this?

I'm going to merge https://github.com/guardian/play-secret-rotation/pull/214 after this is merged, then wait to see how long it takes before Scala Steward raises another PR to update `software.amazon.awssdk:ssm`!
